### DISCUSSION TITLE
How to use InitializeSelfServiceSettingsFlowForBrowsers ... to change password (Go SDK)

### DISCUSSION THREAD

**Discussion by jonnyijapan** [ory/kratos#2583] [Q&A]
In my web backend I'll be needing to talk to Kratos to complete a recovery flow > change password, using a session cookie (not token). I have confirmed that the session cookie is valid for a session using the ToSession SDK function

https://github.com/ory/kratos-client-go/blob/master/docs/V0alpha2Api.md#initializeselfservicesettingsflowforbrowsers

For now, I have something that looks like this.
```
func changePasswordUsingCookie(newPassword string, sessionCookie string, csrfkey string, csrfvalue string, flow string) bool {
	log.Println("changePasswordUsingCookie, flow:", flow)
	success := false

	// From here: https://www.ory.sh/docs/kratos/self-service/flows/user-settings#code-examples-for-nodejs-reactjs-go-

	apiClient := getPublicApiClient()
	ctx := context.Background()

	initRequest := apiClient.V0alpha2Api.InitializeSelfServiceSettingsFlowForBrowsers(ctx)
	settingsflow, settingsflowresponse, settingsflowerror := initRequest.Execute()
	if settingsflowerror != nil {
		log.Println("Failed InitializeSelfServiceSettingsFlowForBrowsers: ", settingsflowerror)
		log.Println("response:", settingsflowresponse)
	} else {
		log.Println("settingsflow:", settingsflow)

		// cookie := "ory_kratos_session=" + sessionCookie + "; " + csrfkey + "=" + csrfvalue
		cookie := "ory_kratos_session=" + sessionCookie
		log.Println("cookie used here:", cookie)

		thing := kratosclient.SubmitSelfServiceSettingsFlowWithPasswordMethodBody{
			Method:   "password",
			Password: newPassword,
			// CsrfToken: &csrfvalue,
		}
		submitBody := kratosclient.SubmitSelfServiceSettingsFlowBody{
			SubmitSelfServiceSettingsFlowWithPasswordMethodBody: &thing,
		}

		// Submit the form
		settingsflow, response, err := apiClient.V0alpha2Api.SubmitSelfServiceSettingsFlow(ctx).Flow(settingsflow.Id).Cookie(cookie).SubmitSelfServiceSettingsFlowBody(submitBody).Execute()

		if err != nil {
			log.Println("Failed SubmitSelfServiceSettingsFlow: ", err)
			log.Println("response:", response)
			log.Println("settingsflow:", settingsflow)
		} else {
			success = true
		}
	}

	return success
}

func getKratosPublicUrl() string {
	return "http://host.docker.internal:4433"
}

func getPublicApiClient() *kratosclient.APIClient {
	configuration := kratosclient.NewConfiguration()
	configuration.Servers = []kratosclient.ServerConfiguration{
		{
			URL: getKratosPublicUrl(),
		},
	}
	apiClient := kratosclient.NewAPIClient(configuration)
	return apiClient
}
```

The problem is with the InitializeSelfServiceSettingsFlowForBrowsers() which upon execution always causes an 401 error. Here are the logs on the web app side:

> development_1  | 2022/07/08 03:38:25 Failed InitializeSelfServiceSettingsFlowForBrowsers:  401 Unauthorized
> development_1  | 2022/07/08 03:38:25 response: &{401 Unauthorized 401 HTTP/1.1 1 1 map[Cache-Control:[private, no-cache, no-store, must-revalidate] Content-Length:[192] Content-Type:[application/json] Date:[Fri, 08 Jul 2022 03:38:25 GMT] Set-Cookie:[csrf_token_806060ca5bf70dff3caa0e5c860002aade9d470a5a4dce73bcfa7ba10778f481=D+PYs0RkRdr9XIHVzPv3jvvGQlhqpM/FUtivkCfYY/8=; Path=/; Max-Age=31536000; HttpOnly; SameSite=Lax] Vary:[Origin Cookie]] {{"error":{"id":"session_inactive","code":401,"status":"Unauthorized","reason":"No active session was found in this request.","message":"request does not have a valid authentication session"}}

**Comment by vinckr** [comment]
Hello @jonnyijapan 
Sorry for the late answer! 
You are trying to initialize the [User and Profile Settings for Server-Side Browser Clients](https://www.ory.sh/docs/kratos/self-service/flows/user-settings#user-and-profile-settings-for-server-side-browser-clients) flow and it fails because no active session is found in the request, correct?

>When using the SDK in a browser app, on the server side you must include the HTTP Cookie Header sent by the client to your server here. This ensures that CSRF and session cookies are respected. (optional)

**Reply by jonnyijapan** [reply]
Yes how can I send cookies with this call? InitializeSelfServiceSettingsFlowForBrowsers 
there seems to be no way of attaching cookies to it

**Reply by armoleon** [reply]
Hello,
I have the same problem and am stuck here. I also couldn't find calling InitializeSelfServiceSettingsFlowForBrowsers method with a valid cookie. Is this a bug or is there another way to do this?

**Reply by DanteTheCreator** [reply]
@vinckr I have the same problem, is there a solution or should I just stop using SDK and go with Axios?

**Reply by DanteTheCreator** [reply]
@zepatrik @aeneasr @icyphox @kevgo @NickUfer 
Can anyone please provide any sort of help on this question?

**Reply by aeneasr** [reply]
Thank you for reminding us to look into this, and very sorry that it has not been resolved earlier. Here is a fix for this problem: https://github.com/ory/kratos/pull/2720
